### PR TITLE
Adding Turkic X-WMT evaluation set for machine translation

### PR DESCRIPTION
**Pull request by mirzakhalov**
This dataset is a human-translated evaluation set for MT crowdsourced and provided by the [Turkic Interlingua ](turkic-interlingua.org) community. It contains eval sets for 8 Turkic languages covering 88 language directions. Languages being covered are:

Azerbaijani (az)
Bashkir (ba)
English (en)
Karakalpak (kaa)
Kazakh (kk)
Kirghiz (ky)
Russian (ru)
Turkish (tr)
Sakha (sah)
Uzbek (uz)

More info about the corpus is here: [https://github.com/turkic-interlingua/til-mt/tree/master/xwmt](https://github.com/turkic-interlingua/til-mt/tree/master/xwmt)

A paper describing the test set is here: [https://arxiv.org/abs/2109.04593](https://arxiv.org/abs/2109.04593)

### CODE REVIEW

File: datasets/turkic_xwmt/README.md [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+  uz-tr: 
+  - uz
+  - tr
+licenses:
+- mit
+multilinguality:
+- translation
+pretty_name: turkic-xwmt
+size_categories:
+- unknown
```
Comment (lhoestq) [inline datasets/turkic_xwmt/README.md:283]:
```suggestion
- n<1K
```

File: datasets/turkic_xwmt/README.md [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+source_datasets: []
+task_categories:
+- conditional-text-generation
+task_ids:
+- machine-translation
+---
+
+# Dataset Card for [Dataset Name]
```
Comment (lhoestq) [inline datasets/turkic_xwmt/README.md:291]:
```suggestion
# Dataset Card for turkic-xwmt
```

File: datasets/turkic_xwmt/README.md [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+  - uz
+  - tr
+licenses:
+- mit
+multilinguality:
+- translation
+pretty_name: turkic-xwmt
+size_categories:
+- unknown
+source_datasets: []
```
Comment (lhoestq) [inline datasets/turkic_xwmt/README.md:284]:
```suggestion
source_datasets:
- "other-WMT 2020 News Translation Task"
```

File: datasets/turkic_xwmt/README.md [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+- [Additional Information](#additional-information)
+  - [Dataset Curators](#dataset-curators)
+  - [Licensing Information](#licensing-information)
+  - [Citation Information](#citation-information)
+  - [Contributions](#contributions)
+
+## Dataset Description
+
+- **Homepage:** [https://github.com/turkic-interlingua/til-mt/tree/master/xwmt](https://github.com/turkic-interlingua/til-mt/tree/master/xwmt)
+- **Repository:**[https://github.com/turkic-interlingua/til-mt/tree/master/xwmt](https://github.com/turkic-interlingua/til-mt/tree/master/xwmt)
```
Comment (lhoestq) [inline datasets/turkic_xwmt/README.md:321]:
```suggestion
- **Homepage:**
- **Repository:**[GitHub](https://github.com/turkic-interlingua/til-mt/tree/master/xwmt)
```

File: datasets/turkic_xwmt/README.md [hunk 5]
```diff
@@ -0,0 +1,501 @@
...
+
+Currently covered languages are (besides English and Russian):
+Azerbaijani (az)
+Bashkir (ba)
+Karakalpak (kaa)
+Kazakh (kk)
+Kirghiz (ky)
+Turkish (tr)
+Sakha (sah)
+Uzbek (uz)
```
Comment (lhoestq) [inline datasets/turkic_xwmt/README.md:344]:
```suggestion
Currently covered languages are (besides English and Russian):
- Azerbaijani (az)
- Bashkir (ba)
- Karakalpak (kaa)
- Kazakh (kk)
- Kirghiz (ky)
- Turkish (tr)
- Sakha (sah)
- Uzbek (uz)
```

File: datasets/turkic_xwmt/README.md [hunk 6]
```diff
@@ -0,0 +1,501 @@
...
+
+## Dataset Structure
+
+[More Information Needed]
```
Comment (lhoestq) [inline datasets/turkic_xwmt/README.md:348]:
```suggestion

```

File: datasets/turkic_xwmt/README.md [hunk 7]
```diff
@@ -0,0 +1,501 @@
...
+
+### Data Instances
+
+```
+A random example from the Russian-Uzbek set:
+{'ru': 'Моника Мутсвангва , министр информации Зимбабве , утверждает , что полиция вмешалась в отъезд Магомбейи из соображений безопасности и вследствие состояния его здоровья .', 'uz': 'Zimbabvening Axborot vaziri , Monika Mutsvanva Magombeyining xavfsizligi va sog'ligi tufayli bo'lgan jo'nab ketishinida politsiya aralashuvini ushlab turadi .'}
+```
+### Data Fields
+
+[More Information Needed]
```
Comment (lhoestq) [inline datasets/turkic_xwmt/README.md:358]:
```suggestion
Each example has one field "translation" that contains two subfields: one per language, e.g. for the Russian-Uzbek set:
- **translation**: a dictionary with two subfields:
  - **ru**: the russian text
  - **uz**: the uzbek text
```

File: datasets/turkic_xwmt/README.md [hunk 8]
```diff
@@ -0,0 +1,501 @@
...
+
+## Dataset Structure
+
+[More Information Needed]
+
+### Data Instances
+
+```
+A random example from the Russian-Uzbek set:
+{'ru': 'Моника Мутсвангва , министр информации Зимбабве , утверждает , что полиция вмешалась в отъезд Магомбейи из соображений безопасности и вследствие состояния его здоровья .', 'uz': 'Zimbabvening Axborot vaziri , Monika Mutsvanva Magombeyining xavfsizligi va sog'ligi tufayli bo'lgan jo'nab ketishinida politsiya aralashuvini ushlab turadi .'}
```
Comment (lhoestq) [inline datasets/turkic_xwmt/README.md:354]: the "translation" parent field is missing

````suggestion
A random example from the Russian-Uzbek set:
```
{"translation": {'ru': 'Моника Мутсвангва , министр информации Зимбабве , утверждает , что полиция вмешалась в отъезд Магомбейи из соображений безопасности и вследствие состояния его здоровья .', 'uz': 'Zimbabvening Axborot vaziri , Monika Mutsvanva Magombeyining xavfsizligi va sog'ligi tufayli bo'lgan jo'nab ketishinida politsiya aralashuvini ushlab turadi .'}}
````

File: datasets/turkic_xwmt/README.md [hunk 9]
```diff
@@ -0,0 +1,501 @@
...
+```
+### Data Fields
+
+[More Information Needed]
+
+### Data Splits
+
+[More Information Needed]
```
Comment (lhoestq) [inline datasets/turkic_xwmt/README.md:362]:
```suggestion

<details>
  <summary>Click here to show the number of examples per configuration:</summary>

|         |   test |
|:--------|-------:|
| az-ba   |    600 |
| az-en   |    600 |
| az-kaa  |    300 |
| az-kk   |    500 |
| az-ky   |    500 |
| az-ru   |    600 |
| az-sah  |    300 |
| az-tr   |    500 |
| az-uz   |    600 |
| ba-az   |    600 |
| ba-en   |   1000 |
| ba-kaa  |    300 |
| ba-kk   |    700 |
| ba-ky   |    500 |
| ba-ru   |   1000 |
| ba-sah  |    300 |
| ba-tr   |    700 |
| ba-uz   |    900 |
| en-az   |    600 |
| en-ba   |   1000 |
| en-kaa  |    300 |
| en-kk   |    700 |
| en-ky   |    500 |
| en-ru   |   1000 |
| en-sah  |    300 |
| en-tr   |    700 |
| en-uz   |    900 |
| kaa-az  |    300 |
| kaa-ba  |    300 |
| kaa-en  |    300 |
| kaa-kk  |    300 |
| kaa-ky  |    300 |
| kaa-ru  |    300 |
| kaa-sah |    300 |
| kaa-tr  |    300 |
| kaa-uz  |    300 |
| kk-az   |    500 |
| kk-ba   |    700 |
| kk-en   |    700 |
| kk-kaa  |    300 |
| kk-ky   |    500 |
| kk-ru   |    700 |
| kk-sah  |    300 |
| kk-tr   |    500 |
| kk-uz   |    700 |
| ky-az   |    500 |
| ky-ba   |    500 |
| ky-en   |    500 |
| ky-kaa  |    300 |
| ky-kk   |    500 |
| ky-ru   |    500 |
| ky-sah  |    300 |
| ky-tr   |    400 |
| ky-uz   |    500 |
| ru-az   |    600 |
| ru-ba   |   1000 |
| ru-en   |   1000 |
| ru-kaa  |    300 |
| ru-kk   |    700 |
| ru-ky   |    500 |
| ru-sah  |    300 |
| ru-tr   |    700 |
| ru-uz   |    900 |
| sah-az  |    300 |
| sah-ba  |    300 |
| sah-en  |    300 |
| sah-kaa |    300 |
| sah-kk  |    300 |
| sah-ky  |    300 |
| sah-ru  |    300 |
| sah-tr  |    300 |
| sah-uz  |    300 |
| tr-az   |    500 |
| tr-ba   |    700 |
| tr-en   |    700 |
| tr-kaa  |    300 |
| tr-kk   |    500 |
| tr-ky   |    400 |
| tr-ru   |    700 |
| tr-sah  |    300 |
| tr-uz   |    600 |
| uz-az   |    600 |
| uz-ba   |    900 |
| uz-en   |    900 |
| uz-kaa  |    300 |
| uz-kk   |    700 |
| uz-ky   |    500 |
| uz-ru   |    900 |
| uz-sah  |    300 |
| uz-tr   |    600 |

</details>

```

File: datasets/turkic_xwmt/turkic_xwmt.py [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+            homepage="https://github.com/turkicinterlingua/til-mt",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        path = dl_manager.download_and_extract(_DATA_URL)
+
+        source, target = self.config.language_pair
+        source_path = f"{path}/test/{source}-{target}/{source}-{target}.{source}.txt"
+        target_path = f"{path}/test/{source}-{target}/{source}-{target}.{target}.txt"
```
Comment (lhoestq) [inline datasets/turkic_xwmt/turkic_xwmt.py:102]: Please use `os.path.join` or you may have issues on windows or in streaming mode:
```suggestion
        source_path = os.path.join(path}, "test", f"{source}-{target}", f"{source}-{target}.{source}.txt")
        target_path = os.path.join(path}, "test", f"{source}-{target}", f"{source}-{target}.{target}.txt")
```

**Review by lhoestq — COMMENTED** [review]
Thanks a lot for adding this dataset ! Good job with the dataset script and the dataset card, they're really good.

Regarding the dummy data, I think we can only keep the dummy data of the `az-ba` configuration, in order to keep things simple, so feel free to delete the other dummy_data.zip files :)

I also left some comments, mostly some formatting stuff

File: datasets/turkic_xwmt/README.md (hunk 1 above)
Comment (mirzakhalov) [reply datasets/turkic_xwmt/README.md]: Done!

File: datasets/turkic_xwmt/README.md (hunk 2 above)
Comment (mirzakhalov) [reply datasets/turkic_xwmt/README.md]: Done!

File: datasets/turkic_xwmt/README.md (hunk 3 above)
Comment (mirzakhalov) [reply datasets/turkic_xwmt/README.md]: Done!

**Comment by mirzakhalov** [comment]
hi! Thank you for all the comments! I believe I addressed them all. Let me know if there is anything else

File: datasets/turkic_xwmt/README.md [hunk 11]
```diff
@@ -0,0 +1,595 @@
...
+- mit
+multilinguality:
+- translation
+pretty_name: turkic_xwmt
+size_categories:
+- n<1K
+task_categories:
+- conditional-text-generation
+task_ids:
+- machine-translation
```
Comment (lhoestq) [inline datasets/turkic_xwmt/README.md:287]:
```suggestion
- machine-translation
source_datasets:
- "other-WMT 2020 News Translation Task"
```

File: datasets/turkic_xwmt/turkic_xwmt.py [hunk 12]
```diff
@@ -0,0 +1,134 @@
...
+
+        return [
+            datasets.SplitGenerator(name=datasets.Split.TEST, gen_kwargs=files["test"]),
+        ]
+
+    def _generate_examples(self, source_file, target_file):
+        """This function returns the examples in the raw (text) form."""
+        source_sentences, target_sentences = None, None
+        source_sentences = open(source_file, "r").read().strip().split("\n")
+        target_sentences = open(target_file, "r").read().strip().split("\n")
```
Comment (lhoestq) [inline datasets/turkic_xwmt/turkic_xwmt.py:120]:
```suggestion
        source_sentences = open(source_file, encoding="utf-8").read().strip().split("\n")
        target_sentences = open(target_file, encoding="utf-8").read().strip().split("\n")
```

File: datasets/turkic_xwmt/README.md [hunk 13]
```diff
@@ -0,0 +1,595 @@
...
+- Kirghiz (ky)
+- Turkish (tr)
+- Sakha (sah)
+- Uzbek (uz)
+
+## Dataset Structure
+
+### Data Instances
+
+```
```
Comment (lhoestq) [inline datasets/turkic_xwmt/README.md:349]:
````suggestion
````

**Review by lhoestq — APPROVED** [review]
Thank you ! It looks all good :)

Let me fix some minor remaining things and I think we can merge

**Comment by mirzakhalov** [comment]
Hi there! I was wondering if there is anything else to change before this can be merged

**Comment by mirzakhalov** [comment]
@lhoestq Hi! Just a gentle reminder about the steps to merge this one!

**Comment by lhoestq** [comment]
Thanks for the heads up ! I think I fixed the last issue with the YAML tags

**Comment by lhoestq** [comment]
The CI failure is unrelated to this PR and fixed on master, let's merge :)

Thanks a lot !

**Pull request merged**